packages/adapter-unstorage/src/index.ts

```diff
@@ -0,0 +1,355 @@
+  options: UnstorageAdapterOptions = {}
+): Adapter {
+  const mergedOptions = {
+    ...defaultOptions,
+    ...options,
+  }
+
+  const { baseKeyPrefix } = mergedOptions
+  const accountKeyPrefix = baseKeyPrefix + mergedOptions.accountKeyPrefix
+  const accountByUserIdPrefix =
+    baseKeyPrefix + mergedOptions.accountByUserIdPrefix
+  const emailKeyPrefix = baseKeyPrefix + mergedOptions.emailKeyPrefix
+  const sessionKeyPrefix = baseKeyPrefix + mergedOptions.sessionKeyPrefix
+  const sessionByUserIdKeyPrefix =
+    baseKeyPrefix + mergedOptions.sessionByUserIdKeyPrefix
+  const userKeyPrefix = baseKeyPrefix + mergedOptions.userKeyPrefix
+  const verificationTokenKeyPrefix =
+    baseKeyPrefix + mergedOptions.verificationTokenKeyPrefix
+
+  async function getItem<T extends StorageValue>(key: string) {
+    if (mergedOptions.useItemRaw) {
+      return await storage.getItemRaw<T>(key)
+    } else {
+      return await storage.getItem<T>(key)
+    }
+  }
+
+  async function setItem(key: string, value: string) {
+    if (mergedOptions.useItemRaw) {
+      return await storage.setItemRaw(key, value)
+    } else {
+      return await storage.setItem(key, value)
+    }
+  }
+
+  const setObjectAsJson = async (key: string, obj: any) => {
+    if (mergedOptions.useItemRaw) {
+      await storage.setItemRaw(key, obj)
+    } else {
+      await storage.setItem(key, JSON.stringify(obj))
+    }
+  }
+
+  const setAccount = async (id: string, account: AdapterAccount) => {
+    const accountKey = accountKeyPrefix + id
+    await setObjectAsJson(accountKey, account)
+    await setItem(accountByUserIdPrefix + account.userId, accountKey)
+    return account
+  }
+
+  const getAccount = async (id: string) => {
+    const account = await getItem<AdapterAccount>(accountKeyPrefix + id)
+    if (!account) return null
+    return hydrateDates(account)
+  }
+
+  const setSession = async (
+    id: string,
+    session: AdapterSession
+  ): Promise<AdapterSession> => {
+    const sessionKey = sessionKeyPrefix + id
+    await setObjectAsJson(sessionKey, session)
+    await setItem(sessionByUserIdKeyPrefix + session.userId, sessionKey)
+    return session
+  }
+
+  const getSession = async (id: string) => {
+    const session = await getItem<AdapterSession>(sessionKeyPrefix + id)
+    if (!session) return null
+    return hydrateDates(session)
+  }
+
+  const setUser = async (
+    id: string,
+    user: AdapterUser
+  ): Promise<AdapterUser> => {
+    await setObjectAsJson(userKeyPrefix + id, user)
+    await setItem(`${emailKeyPrefix}${user.email as string}`, id)
+    return user
+  }
+
+  const getUser = async (id: string) => {
+    const user = await getItem<AdapterUser>(userKeyPrefix + id)
+    if (!user) return null
+    return hydrateDates(user)
+  }
+
+  return {
+    async createUser(user) {
+      const id = crypto.randomUUID()
+      // TypeScript thinks the emailVerified field is missing
+      // but all fields are copied directly from user, so it's there
```

Comment:
Is this correct?

Reply:
fix: https://github.com/nextauthjs/next-auth/pull/9501/commits/88d1b19d1e4ca7a060c1bfc460db29b0ce99b3bb